src/app/dashboard/(admin)/admin/(settings)/settings/account/_component.tsx/password.tsx

```diff
@@ -1,13 +1,72 @@
 "use client";
 
-import { useState } from "react";
+import axios from "axios";
+import { SetStateAction, useState } from "react";
 
 import CustomButton from "~/components/common/common-button/common-button";
 import CustomInput from "~/components/common/input/input";
 import PasswordSuccessfulModal from "~/components/common/modals/password-successful";
 
 const Password = () => {
   const [open, setOpen] = useState<boolean>(false);
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [confirmNewPassword, setConfirmNewPassword] = useState("");
+  const [error, setError] = useState<string>("");
+
+  const [isLengthValid, setIsLengthValid] = useState(false);
+  const [hasUppercase, setHasUppercase] = useState(false);
+  const [hasNumber, setHasNumber] = useState(false);
+
+  const handleNewPasswordChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const password = event.target.value;
+    setNewPassword(password);
+
+    setIsLengthValid(password.length >= 8);
+    setHasUppercase(/[A-Z]/.test(password));
+    setHasNumber(/\d/.test(password));
+  };
+
+  const handleChangePassword = async () => {
+    if (!isLengthValid || !hasUppercase || !hasNumber) {
+      setError("New password does not meet all the criteria.");
+      return;
+    }
+
+    if (newPassword !== confirmNewPassword) {
+      setError("Password does not match");
+      return;
+    }
+
+    try {
+      await axios.patch(
+        "https://deployment.api-expressjs.boilerplate.hng.tech/api/v1/auth/change-password",
```

Comment:
there is a util to get base url not this

you are working across all teams not just one